pkg/controllers/interruption/events/events.go

```diff
@@ -19,65 +19,138 @@ import (
+			Type:           v1.EventTypeWarning,
+			Reason:         "Stopping",
+			Message:        fmt.Sprintf("Machine %s event: Machine is stopping", machine.Name),
+			DedupeValues:   []string{machine.Name},
+		},
+	}
+	if node != nil {
+		evts = append(evts, events.Event{
+			InvolvedObject: node,
+			Type:           v1.EventTypeWarning,
+			Reason:         "Stopping",
+			Message:        fmt.Sprintf("Node %s event: Node is stopping", node.Name),
+			DedupeValues:   []string{node.Name},
+		})
 	}
+	return evts
 }
 
-func InstanceTerminating(node *v1.Node) events.Event {
-	return events.Event{
-		InvolvedObject: node,
-		Type:           v1.EventTypeWarning,
-		Reason:         "InstanceTerminating",
-		Message:        fmt.Sprintf("Node %s event: Instance is terminating", node.Name),
-		DedupeValues:   []string{node.Name},
+func Terminating(node *v1.Node, machine *v1alpha5.Machine) []events.Event {
+	evts := []events.Event{
+		{
+			InvolvedObject: machine,
+			Type:           v1.EventTypeWarning,
+			Reason:         "Terminating",
+			Message:        fmt.Sprintf("Machine %s event: Machine is terminating", machine.Name),
+			DedupeValues:   []string{machine.Name},
+		},
 	}
+	if node != nil {
+		evts = append(evts, events.Event{
+			InvolvedObject: node,
+			Type:           v1.EventTypeWarning,
+			Reason:         "Terminating",
+			Message:        fmt.Sprintf("Node %s event: Node is terminating", node.Name),
+			DedupeValues:   []string{node.Name},
+		})
+	}
+	return evts
 }
 
-func InstanceUnhealthy(node *v1.Node) events.Event {
-	return events.Event{
-		InvolvedObject: node,
-		Type:           v1.EventTypeWarning,
-		Reason:         "InstanceUnhealthy",
-		Message:        fmt.Sprintf("Node %s event: An unhealthy warning was triggered for the node", node.Name),
-		DedupeValues:   []string{node.Name},
+func Unhealthy(node *v1.Node, machine *v1alpha5.Machine) []events.Event {
+	evts := []events.Event{
+		{
+			InvolvedObject: machine,
+			Type:           v1.EventTypeWarning,
+			Reason:         "Unhealthy",
+			Message:        fmt.Sprintf("Machine %s event: An unhealthy warning was triggered for the machine", machine.Name),
+			DedupeValues:   []string{machine.Name},
+		},
+	}
+	if node != nil {
+		evts = append(evts, events.Event{
+			InvolvedObject: node,
+			Type:           v1.EventTypeWarning,
+			Reason:         "Unhealthy",
+			Message:        fmt.Sprintf("Node %s event: An unhealthy warning was triggered for the node", node.Name),
+			DedupeValues:   []string{node.Name},
+		})
 	}
+	return evts
 }
 
-func NodeTerminatingOnInterruption(node *v1.Node) events.Event {
-	return events.Event{
-		InvolvedObject: node,
-		Type:           v1.EventTypeWarning,
-		Reason:         "NodeTerminatingOnInterruption",
-		Message:        fmt.Sprintf("Node %s event: Interruption triggered termination for the node", node.Name),
-		DedupeValues:   []string{node.Name},
+func TerminatingOnInterruption(node *v1.Node, machine *v1alpha5.Machine) []events.Event {
```

Comment:
This seems spammy that we'd emit events for both node and machine.

Reply:
I agree that firing two events for a single action could be seen as spammy but we need to cut the line between making sure that node events are still observable for users and making sure that we are also surfacing events for Machines (since the actions are taking place against them). Maybe let's discuss a good way to delineate which events should go where